:farmer: TestROS2TopicUseSimTime.test_pub_times failing in Linux Coverage

## Bug report

**Required Info:**

- Operating System:
  - Ubuntu Noble
- Installation type:
  - Source
- Version or commit hash:
  - Rolling

#### Steps to reproduce issue
1. Run a build in [nightly_linux_coverage](https://ci.ros2.org/job/nightly_linux_coverage/)
2. See test ros2topic.ros2topic.test.test_use_sim_time.test_use_sim_time fail

#### Additional information

Reference build:
* [nightly_linux_coverage#2511](https://ci.ros2.org/job/nightly_linux_coverage/2511)

Test regression:
* [ros2topic.ros2topic.test.test_use_sim_time.test_use_sim_time](https://ci.ros2.org/job/nightly_linux_coverage/2511/testReport/junit/ros2topic.ros2topic.test/test_use_sim_time/test_use_sim_time/)

Log output:
```
FAIL: test_use_sim_time.TestROS2TopicUseSimTime.test_pub_times
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins-agent/workspace/nightly_linux_coverage/ws/install/launch_testing/lib/python3.12/site-packages/launch_testing/markers.py", line 61, in _wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jenkins-agent/workspace/nightly_linux_coverage/ws/src/ros2/ros2cli/ros2topic/test/test_use_sim_time.py", line 124, in test_pub_times
    assert command.wait_for_shutdown(timeout=5)
AssertionError: assert False
 +  where False = <bound method ProcessProxy.wait_for_shutdown of <launch_testing.tools.process.ProcessProxy object at 0x7f2122599580>>(timeout=5)
 +    where <bound method ProcessProxy.wait_for_shutdown of <launch_testing.tools.process.ProcessProxy object at 0x7f2122599580>> = <launch_testing.tools.process.ProcessProxy object at 0x7f2122599580>.wait_for_shutdown
```

--- a/ros2topic/test/test_use_sim_time.py
+++ b/ros2topic/test/test_use_sim_time.py
@@ -97,7 +97,7 @@ class TestROS2TopicUseSimTime(unittest.TestCase):
         # test method:
         #   speed up sim time to half second, then expect the process
         #   will end at half time compare to system time (5s).
-        
+
         # Start the timer from 1 second, so that 1st message fires event.
         self.clock_sec = 1
         # Let's speed up 2x fast

--- a/ros2topic/test/test_use_sim_time.py
+++ b/ros2topic/test/test_use_sim_time.py
@@ -15,6 +15,7 @@
 import functools
 import re
 import sys
+import time
 import unittest
 
 from launch import LaunchDescription
@@ -78,6 +79,14 @@ class TestROS2TopicUseSimTime(unittest.TestCase):
         self.publisher.publish(msg)
         self.clock_sec += 1
 
+    def wait_for_command_online(self):
+        max_seconds_to_wait = 5
+        end_time = time.time() + max_seconds_to_wait
+        while self.publisher.get_subscription_count() < 1:
+            time.sleep(0.1)
+            assert time.time() <= end_time  # timeout waiting for command online
+        assert self.publisher.get_subscription_count() >= 1
+
     def setUp(self):
         self.context = rclpy.context.Context()
         rclpy.init(context=self.context)
@@ -119,6 +128,8 @@ class TestROS2TopicUseSimTime(unittest.TestCase):
                     filtered_rmw_implementation=get_rmw_implementation_identifier()
                 )
             ) as command:
+                # Check if the command line is ready to receive /clock messages
+                self.wait_for_command_online()
                 # The process will end up in around 2.5s (here we set 3s)
                 self.executor.spin_until_future_complete(
                     rclpy.task.Future(), timeout_sec=3
@@ -184,6 +195,8 @@ class TestROS2TopicUseSimTime(unittest.TestCase):
                     filtered_rmw_implementation=get_rmw_implementation_identifier()
                 )
             ) as command:
+                # Check if the command line is ready to receive /clock messages
+                self.wait_for_command_online()
                 # The future won't complete - we will hit the timeout
                 self.executor.spin_until_future_complete(
                     rclpy.task.Future(), timeout_sec=5
@@ -241,6 +254,8 @@ class TestROS2TopicUseSimTime(unittest.TestCase):
                     filtered_rmw_implementation=get_rmw_implementation_identifier()
                 )
             ) as command:
+                # Check if the command line is ready to receive /clock messages
+                self.wait_for_command_online()
                 # The future won't complete - we will hit the timeout
                 self.executor.spin_until_future_complete(
                     rclpy.task.Future(), timeout_sec=10

--- a/ros2topic/test/test_use_sim_time.py
+++ b/ros2topic/test/test_use_sim_time.py
@@ -97,7 +97,9 @@ class TestROS2TopicUseSimTime(unittest.TestCase):
         # test method:
         #   speed up sim time to half second, then expect the process
         #   will end at half time compare to system time (5s).
-        self.clock_sec = 0
+        
+        # Start the timer from 1 second, so that 1st message fires event.
+        self.clock_sec = 1
         # Let's speed up 2x fast
         rate = 0.5
         clock_timer = self.node.create_timer(rate, self.timer_callback)

